### PR TITLE
Integration RequestController to VR Plugin

### PR DESCRIPTION
**Pull request by KKarlash**
Please take a look at review for integration RequestController to VR plugin.

@DKlimenko 
@VladSemenyuk
@bonewell 
@giangtqh 
@thinhts

### CODE REVIEW

**Comment by KKarlash** [comment]
Files cproject and project will be reverted

**Comment by giangtqh** [comment]
reviewed

**Comment by VladSemenyuk** [comment]
Reviewed

**Comment by thinhts** [comment]
reviewed

**Comment by DKlimenko** [comment]
reviewed

File: src/components/vr_cooperation/include/vr_cooperation/vr_module_timer.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+ private:
+  functional_modules::TimeUnit custom_interval_;
+  uint32_t app_id_;
+  uint32_t correlation_id_;
+};
+
+}  // namespace can_cooperation
+#endif /* SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_VR_MODULE_TIMER_H_ */
```
Comment (bonewell) [inline src/components/vr_cooperation/include/vr_cooperation/vr_module_timer.h]: Incorrect type of comment. Should be `//`

File: src/components/vr_cooperation/src/request_controller.cc [hunk 2]
```diff
@@ -31,25 +31,54 @@
  */
 
 #include "vr_cooperation/request_controller.h"
+#include "utils/logger.h"
+#include "functional_module/timer/timer_director_impl.h"
+#include "functional_module/settings.h"
```
Comment (bonewell) [inline src/components/vr_cooperation/src/request_controller.cc]: Includes should be sort by alphabet.

File: src/components/vr_cooperation/src/request_controller.cc [hunk 3]
```diff
@@ -31,25 +31,54 @@
...
 
 namespace vr_cooperation {
 namespace request_controller {
 
 CREATE_LOGGERPTR_GLOBAL(logger_, "VRRequestController")
 
 RequestController::RequestController() {
+  functional_modules::TimeUnit timeout_seconds = 100;
```
Comment (bonewell) [inline src/components/vr_cooperation/src/request_controller.cc]: Use const.

File: src/components/vr_cooperation/src/request_controller.cc [hunk 4]
```diff
@@ -31,25 +31,54 @@
...
+  mobile_request_list_.erase(mobile_correlation_id);
+  timer_.RemoveTrackable(TrackableMessage(0, mobile_correlation_id));
+}
+
+void RequestController::OnTimeoutTriggered(const TrackableMessage& expired) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::map<correlation_id, MobileRequestPtr>::iterator it =
+    mobile_request_list_.find(expired.correlation_id());
+  if (mobile_request_list_.end() == it) {
+    // no corresponding request found, error.
```
Comment (bonewell) [inline src/components/vr_cooperation/src/request_controller.cc]: Write log for this case.

File: src/components/vr_cooperation/include/vr_cooperation/request_controller.h [hunk 5]
```diff
@@ -81,8 +82,15 @@ class RequestController {
...
   void DeleteRequest(const uint32_t& mobile_correlation_id);
 
+  /**
+   * @brief Removes messages from map by timer
+   * @param TrackableMessage expired message
+   */
+  void OnTimeoutTriggered(const TrackableMessage& expired);
+
  private:
   std::map<correlation_id, MobileRequestPtr> mobile_request_list_;
```
Comment (bonewell) [inline src/components/vr_cooperation/include/vr_cooperation/request_controller.h]: It will be good to use typedef.

**Comment by bonewell** [comment]
Reviewed.

**Pull request merged**